### ISSUE TITLE
Leading zero in id

### ISSUE THREAD

**Issue by ZCKevin-TW** [NTU-SP/SP-Hw1-release#11]
Hi, in input checking section, nothing is mentioned about leading zero.
1. will leading zero appear in testdata
2. should leading zero be valid for numbers? (e.g. Is id = 000902001 valid?)

**Comment by imingasco** [comment]
No leading zero appear in testdata.

**Comment by ZCKevin-TW** [comment]
How about leading spaces and trailing spaces?
Is it guaranteed that when client connects to read_server, valid input always contains no leading spaces and no trailing spaces then end of line?

**Comment by imingasco** [comment]
Yes, no leading/trailing spaces appear in testdata(no matter valid or invalid).
